src/graphql/schema/User/typeDefs.gql

```diff
@@ -3,25 +3,38 @@ extend type Query {
   allUsers: [User]!
 }
 
+# TODO refactor to take userid from auth token
 extend type Mutation {
-  updateUser(input: updateUserInput!): updatedUser!
+  updateUser(input: updateUserInput!): User # TODO put user id in separate argument e.g. updateUser(id: ID!, update: updateUserInput): User!
+  createUser(input: createUserInput!): User
 }
 
 type User {
-  id: ID!
-  email: String
+  email: String!
   avatarURL: String
   firstName: String
   lastName: String
   trips: [Trip]
 }
 
+input UserInput {
+  email: String!
+  avatarURL: String
+  firstName: String
+  lastName: String
+}
+
 input updateUserInput {
   id: ID!
-  name: String
+  firstName: String
+  lastName: String
+  email: String
+  avatarURL: String
 }
 
-type updatedUser {
-  id: ID!
-  name: String
+input createUserInput {
+  email: String!
+  avatarURL: String
+  firstName: String!
```

Comment:
Do the names have to be mandatory?
Does a user when he invites friends then neccessarily need to give their first and last names?
Firstname in this case would be enough for me

Reply:
fair point. to me, the `createUser` mutation is going to replaced by a `register` mutation anyway once we integrate authentication. but from a UX perspective making first name mandatory would make a lot of sense, i think.